### PR TITLE
[Docs] Small updates to getrawtransaction description

### PR DESCRIPTION
**Pull request by amitiuttarwar**
As per review comments on https://github.com/bitcoin/bitcoin/pull/15159

### CODE REVIEW

**Comment by amitiuttarwar** [comment]
Question about whitespace 😝-
There seems to be an implicit convention of indenting the first line(s) of the `RPCHelpMan` descriptions and I'm wondering why? The changes look recent but I didn't find anything relevant in the style guide. If this is intentional, maybe I can add something there? 

Ex: https://github.com/bitcoin/bitcoin/blob/master/src/wallet/rpcdump.cpp#L113 & https://github.com/bitcoin/bitcoin/blob/master/src/wallet/rpcdump.cpp#L1196

However, this one doesn't extra-indent the first line: 
https://github.com/bitcoin/bitcoin/blob/master/src/wallet/rpcdump.cpp#L755

**Comment by MarcoFalke** [comment]
There is no guideline about whitespace as long as you stick to either rule-of-thumb:

* install `clang-format` and run the https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy script
* Keep the formatting as is

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -67,13 +67,16 @@ static UniValue getrawtransaction(const JSONRPCRequest& request)
...
-                "is in a block in the blockchain.\n"
+                "\nReturn the raw transaction data.\n"
+
+            "\nBy default this function only works for mempool transactions. When called with a blockhash\n"
+            "argument, getrawtransaction will return the transaction if the specified block is available and\n"
+            "the transaction is found in that block. When called without a blockhash argument, getrawtransaction\n"
+            "will return the transaction if it is in the mempool, or if -txindex is enabled and the transaction\n"
+            "is in a block in the blockchain.\n"
+
+            "\nThe gettransaction RPC can be used to retrieve wallet transactions.\n"
```
Comment (MarcoFalke) [inline src/rpc/rawtransaction.cpp]: Could add something like

```cpp
                "\nHint: use getmempoolentry to fetch a specific transaction from the mempool.\n",

**Comment by MarcoFalke** [comment]
ACK

**Comment by amitiuttarwar** [comment]
hmm, ok went with format script if indentation was not intentional. added getmempoolentry reference.

**Comment by kristapsk** [comment]
utACK 35c7eb045f79b5a556c9d17eddf3bf59d0d1b55f

File: doc/release-notes.md [hunk 2]
```diff
@@ -254,11 +254,11 @@ in the Low-level Changes section below.
 
 - See the [Mining](#mining) section for changes to `getblocktemplate`.
 
-- The `getrawtransaction` RPC no longer checks the unspent UTXO set for
-  a transaction. The remaining behaviors are as follows: 1. If a
-  blockhash is provided, check the corresponding block. 2. If no
-  blockhash is provided, check the mempool. 3. If no blockhash is
-  provided but txindex is enabled, also check txindex.
+- The `getrawtransaction` RPC & REST endpoint no longer checks the
```
Comment (fanquake) [inline doc/release-notes.md]: nit: RPC & REST endpoints no longer check

**Comment by amitiuttarwar** [comment]
This is ready.

**Review by jnewbery — COMMENTED** [review]
ACK 47012391ec25eb76669c7ba2aa458c3ac611289c.

One nit inline, which could be ignored.

File: doc/release-notes.md [hunk 3]
```diff
@@ -254,11 +254,11 @@ in the Low-level Changes section below.
 
 - See the [Mining](#mining) section for changes to `getblocktemplate`.
 
-- The `getrawtransaction` RPC no longer checks the unspent UTXO set for
-  a transaction. The remaining behaviors are as follows: 1. If a
-  blockhash is provided, check the corresponding block. 2. If no
-  blockhash is provided, check the mempool. 3. If no blockhash is
-  provided but txindex is enabled, also check txindex.
+- The `getrawtransaction` RPC & REST endpoints no longer check the
+  unspent UTXO set for a transaction. The remaining behaviors are as
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply doc/release-notes.md]: I'll let others comment. ACK either way.

**Pull request merged**